controllers/managedocs_controller.go

```diff
@@ -1050,6 +1079,63 @@ func (r *ManagedOCSReconciler) reconcileRookCephOperatorConfig() error {
 	return nil
 }
 
+func (r *ManagedOCSReconciler) reconcileEgressNetworkPolicy() error {
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.egressNetworkPolicy, func() error {
+		if err := r.own(r.egressNetworkPolicy); err != nil {
+			return err
+		}
+		desired := templates.EgressNetworkPolicyTemplate.DeepCopy()
+
+		if r.deadMansSnitchSecret.UID == "" {
+			if err := r.get(r.deadMansSnitchSecret); err != nil {
+				return fmt.Errorf("Unable to get DeadMan's Snitch secret: %v", err)
+			}
+		}
+		dmsURL := string(r.deadMansSnitchSecret.Data["SNITCH_URL"])
+		if dmsURL == "" {
+			return fmt.Errorf("DeadMan's Snitch secret does not contain a SNITCH_URL entry")
+		}
+		snitchURL, err := url.Parse(string(r.deadMansSnitchSecret.Data["SNITCH_URL"]))
+		if err != nil {
+			return fmt.Errorf("Unable to parse DMS url: %v", err)
+		}
+
+		if r.smtpSecret.UID == "" {
+			if err := r.get(r.smtpSecret); err != nil {
+				return fmt.Errorf("Unable to get SMTP secret: %v", err)
+			}
+		}
+		smtpHost := string(r.smtpSecret.Data["host"])
+		if smtpHost == "" {
+			return fmt.Errorf("smtp secret does not contain a host entry")
+		}
+
+		desired.Spec.Egress[0].To.DNSName = snitchURL.Hostname()
+		desired.Spec.Egress[1].To.DNSName = smtpHost
```

Comment:
Addressing Egress rules by an index is very fregile. Especially on template changes.  
As these are not named objects my suggestion will be to have all static egress rules in the template and just append the dynamic ones.

So instead of doing:
```
desired.Spec.Egress[0].To.DNSName = snitchURL.Hostname()
```
you do
```
desired.Spec.Egress = append(desired.Spec.Egress, <new egress object with the proper hostname set>)